### PR TITLE
Fixes File Manager crash when browsing to subfolders in /proc

### PR DESCRIPTION
**Pull request by rehaby**
Fixes a crash when expanding /proc in the side panel of the File Manager.

https://github.com/SerenityOS/serenity/issues/13435

### CODE REVIEW

**Comment by timschumi** [comment]
Haven't tested it yet, but the patch looks reasonable.

There is a similar issue in `Userland/Services/LaunchServer/Launcher.cpp`, introduced in the [same commit](https://github.com/SerenityOS/serenity/commit/4a57be824cc6d4fed95f182431ba807b2327395e). Maybe it's worth it to also fix that one (in a separate commit) while we're at it, even though it's in a different component and not technically related to the linked issue.

File: Userland/Libraries/LibGUI/FileSystemModel.cpp [hunk 1]
```diff
@@ -62,12 +62,13 @@ bool FileSystemModel::Node::fetch_data(String const& full_path, bool is_root)
 
     if (S_ISLNK(mode)) {
         auto sym_link_target_or_error = Core::File::read_link(full_path);
-        if (sym_link_target_or_error.is_error())
-            perror("readlink");
-
-        symlink_target = sym_link_target_or_error.release_value();
-        if (symlink_target.is_null())
+        if (sym_link_target_or_error.is_error()) {
```
Comment (dayarthvader) [inline Userland/Libraries/LibGUI/FileSystemModel.cpp:65]: I am new here too but may I suggest using `TRY(Core::File::read_link(full_path));` This is more idiomatic and achieves the same objective
[TRY](https://github.com/SerenityOS/serenity/blob/HEAD/AK/Try.h#L12
)

Comment (timschumi) [reply Userland/Libraries/LibGUI/FileSystemModel.cpp:65]: The function doesn't return an `ErrorOr<T>`, so `TRY()` won't work out-of-the-box. You would have to refactor the function a bit to make use of that.

**Comment by rehaby** [comment]
@timschumi I have added the extra commit that you talked about. I wasn't able to test it, as I don't really know what the LauncherServer is

**Review by timschumi — CHANGES_REQUESTED** [review]
The PR needs a rebase, it conflicts on... _something_.

Also, you should probably rename the PR to something more fitting, as we are no longer only handling FileManager. What about something like `Userland: Fix unintentional error fallthrough after Core::File::read_link`?

**Pull request closed without merging**